File: coding/14th May 2025/Maximum Ascending Subarray sum.java
//Maximum Ascending Subarray sum
class Solution 
{
    public int maxAscendingSum(int[] nums) 
    {
        int s = nums[0];
        int max = s;
        for(int i = 1; i < nums.length; i++)
        {
            if(nums[i - 1] < nums[i])
                s += nums[i];
            else
                s = nums[i];
            if(s > max)
                max = s;
        }
        return max;
    }
}

File: Coding in Python/19th January 2026/Maximum Ascending Subarray Sum.py
#Maximum Ascending Subarray Sum
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        sum1 = nums[0]
        max1 = sum1
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                sum1 += nums[i]
            else:
                sum1 = nums[i]
            if max1 < sum1:
                max1 = sum1
        return max1
